staging/src/k8s.io/client-go/util/workqueue/default_rate_limiters.go

```diff
@@ -62,6 +62,54 @@ func (r *BucketRateLimiter) NumRequeues(item interface{}) int {
 func (r *BucketRateLimiter) Forget(item interface{}) {
 }
 
+// ItemBucketRateLimiter implements a workqueue ratelimiter API using standard rate.Limiter.
+// Each key is using a separate limiter.
+type ItemBucketRateLimiter struct {
+	r     rate.Limit
+	burst int
+
+	limitersLock sync.Mutex
+	limiters     map[interface{}]*rate.Limiter
+}
+
+var _ RateLimiter = &ItemBucketRateLimiter{}
+
+// NewItemBucketRateLimiter creates new ItemBucketRateLimiter instance.
```

Comment:
What is the rationale for using this kind of rate limit for erroring items?

Reply:
My intent was not to use this for erroring items, rather for success case.

I know that it might be slightly misleading. Do you have idea how to improve this?

I was thinking about:
* adding a comment
* maybe removing this from this file (the fact that it implements RateLimiter suggests that this can be used for error case)

Any other idea? WDYT?

Reply:
I think that adding a comment should be enough - just descibe what are the usecases that we expect it to be used at (it's roughly Daniel`s comment from above: https://github.com/kubernetes/kubernetes/pull/88261/files/6846a0a6b62725a5888d66e2789f2af4fb172045#r380980162 )